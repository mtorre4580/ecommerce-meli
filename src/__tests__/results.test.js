import React from 'react';
import Results from '../components/results/results';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

describe('<Results />', () => {

    it('should render snapshot', () => {
        const data = {author:{name:'Matias Daniel',lastname:'Torre'},categories:['Electrónica, Audio y Video','Audio','Reproductores Portátiles','iPod','iPod classic','80 GB'],items:[{id:'MLA772322281',title:'Ipod Touch 4ta Generacion 32 Gb',price:{currency:'ARS',amount:680,decimals:null},picture:'http://mla-s2-p.mlstatic.com/738532-MLA25881108346_082017-I.jpg',condition:'used',location:'Capital Federal'},{id:'MLA766877875',title:'Ipod Classic 60gb',price:{currency:'ARS',amount:1250,decimals:null},picture:'http://mla-s2-p.mlstatic.com/947045-MLA29185741092_012019-I.jpg',condition:'used',location:'Buenos Aires'},{id:'MLA777676864',title:'Ipod Classic 30gb',price:{currency:'ARS',amount:1000,decimals:null},picture:'http://mla-s1-p.mlstatic.com/809984-MLA29730910940_032019-I.jpg',condition:'used',location:'Buenos Aires'},{id:'MLA769088925',title:'Ipod Touch 5g 32gb',price:{currency:'ARS',amount:3200,decimals:null},picture:'http://mla-s2-p.mlstatic.com/720812-MLA25738240210_072017-I.jpg',condition:'used',location:'Buenos Aires'}]};
        const loading = false;
        const error = false;
        const component = renderer.create(<Router><Results loading={loading} data={data} error={error} /></Router>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

 });



