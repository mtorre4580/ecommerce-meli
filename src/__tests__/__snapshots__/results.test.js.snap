// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Results /> should render snapshot 1`] = `
<div
  className="container"
>
  <section>
    <nav>
      <ol
        className="breadcrumb"
      >
        <li
          className="breadcrumb-item"
        >
          Electrónica, Audio y Video
        </li>
        <li
          className="breadcrumb-item"
        >
          Audio
        </li>
        <li
          className="breadcrumb-item"
        >
          Reproductores Portátiles
        </li>
        <li
          className="breadcrumb-item"
        >
          iPod
        </li>
        <li
          className="breadcrumb-item"
        >
          iPod classic
        </li>
        <li
          className="breadcrumb-item"
        >
          80 GB
        </li>
      </ol>
    </nav>
    <article
      className="item"
    >
      <a
        aria-current={false}
        className="item-image"
        href="items/MLA772322281"
        onClick={[Function]}
      >
        <picture
          className="item-image"
        >
          <img
            alt="Ipod Touch 4ta Generacion 32 Gb"
            className="d-block item-img"
            src="http://mla-s2-p.mlstatic.com/738532-MLA25881108346_082017-I.jpg"
          />
        </picture>
      </a>
      <div
        className="item-ctn"
      >
        <div
          className="item-header"
        >
          <div
            className="d-flex"
          >
            <p
              className="price"
            >
              <span>
                $
              </span>
              680
            </p>
          </div>
          <p
            className="text-location"
          >
            Capital Federal
          </p>
        </div>
        <a
          aria-current={false}
          className="item-title"
          href="items/MLA772322281"
          onClick={[Function]}
        >
          <h2>
            Ipod Touch 4ta Generacion 32 Gb
          </h2>
        </a>
      </div>
    </article>
    <article
      className="item"
    >
      <a
        aria-current={false}
        className="item-image"
        href="items/MLA766877875"
        onClick={[Function]}
      >
        <picture
          className="item-image"
        >
          <img
            alt="Ipod Classic 60gb"
            className="d-block item-img"
            src="http://mla-s2-p.mlstatic.com/947045-MLA29185741092_012019-I.jpg"
          />
        </picture>
      </a>
      <div
        className="item-ctn"
      >
        <div
          className="item-header"
        >
          <div
            className="d-flex"
          >
            <p
              className="price"
            >
              <span>
                $
              </span>
              1,250
            </p>
          </div>
          <p
            className="text-location"
          >
            Buenos Aires
          </p>
        </div>
        <a
          aria-current={false}
          className="item-title"
          href="items/MLA766877875"
          onClick={[Function]}
        >
          <h2>
            Ipod Classic 60gb
          </h2>
        </a>
      </div>
    </article>
    <article
      className="item"
    >
      <a
        aria-current={false}
        className="item-image"
        href="items/MLA777676864"
        onClick={[Function]}
      >
        <picture
          className="item-image"
        >
          <img
            alt="Ipod Classic 30gb"
            className="d-block item-img"
            src="http://mla-s1-p.mlstatic.com/809984-MLA29730910940_032019-I.jpg"
          />
        </picture>
      </a>
      <div
        className="item-ctn"
      >
        <div
          className="item-header"
        >
          <div
            className="d-flex"
          >
            <p
              className="price"
            >
              <span>
                $
              </span>
              1,000
            </p>
          </div>
          <p
            className="text-location"
          >
            Buenos Aires
          </p>
        </div>
        <a
          aria-current={false}
          className="item-title"
          href="items/MLA777676864"
          onClick={[Function]}
        >
          <h2>
            Ipod Classic 30gb
          </h2>
        </a>
      </div>
    </article>
    <article
      className="item"
    >
      <a
        aria-current={false}
        className="item-image"
        href="items/MLA769088925"
        onClick={[Function]}
      >
        <picture
          className="item-image"
        >
          <img
            alt="Ipod Touch 5g 32gb"
            className="d-block item-img"
            src="http://mla-s2-p.mlstatic.com/720812-MLA25738240210_072017-I.jpg"
          />
        </picture>
      </a>
      <div
        className="item-ctn"
      >
        <div
          className="item-header"
        >
          <div
            className="d-flex"
          >
            <p
              className="price"
            >
              <span>
                $
              </span>
              3,200
            </p>
          </div>
          <p
            className="text-location"
          >
            Buenos Aires
          </p>
        </div>
        <a
          aria-current={false}
          className="item-title"
          href="items/MLA769088925"
          onClick={[Function]}
        >
          <h2>
            Ipod Touch 5g 32gb
          </h2>
        </a>
      </div>
    </article>
  </section>
</div>
`;
